pkg/models/order.go

```
@@ -99,3 +100,8 @@ func FetchOrder(db *pop.Connection, user User, id uuid.UUID) (Order, error) {
 
 	return order, nil
 }
+
+// CreateNewMove creates a move associated with these Orders
+func (o *Order) CreateNewMove(db *pop.Connection, logger *zap.Logger, moveType *internalmessages.SelectedMoveType) (*Move, bool) {
```

Take a look at FetchBackupContact in models/backup_contact.go

I think that rather than returning a simple bool here, you could return the err and respond to it from the handler. That way you wouldn't have to pass the logger all the way down here.

---

you could create a new model error to capture the unique constraint error the way we return ErrFetchNotFound in FetchBackupContact.

---

Sure, that helps with the error handling but what to do about the validation errors. If those are the problem you will end up throwing them away without the logger.